pkg/kubelet/collector/interfaces.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+type Interface interface {
+	// Starts the collector process if necessary, e.g., the builtin cadvisor
+	// Otherwise, does nothing and just returns
+	Start() error
+
+	// Returns the basic machine spec
+	// Currently used to report to k8s master the basic information about a node
```

Comment:
I'm wary of documenting use cases on functions, since it's so easy for them to become out of date. If you want, you can make it an example instead. (`e.g. to report to k8s master the basic information about a node`)